Makefile

```diff
@@ -1,66 +1,76 @@
-# Copyright 2015 The Prometheus Authors
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# Copyright 2019 The Caicloud Authors.
 #
-# http://www.apache.org/licenses/LICENSE-2.0
+# The old school Makefile, following are required targets. The Makefile is written
+# to allow building multiple binaries. You are free to add more targets or change
+# existing implementations, as long as the semantics are preserved.
+#
+#   make              - default to 'build' target
+#   make lint         - code analysis with golangci-lint
+#   make test         - run unit test
+#   make build        - build binary in a Golang container
+#   make build-local  - build local binary
+#   make container    - build container
+#   make push         - push container
+#   make clean        - clean up
+#
+# The makefile is also responsible to populate project version information.
 #
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-GO    := GO15VENDOREXPERIMENT=1 go
-PROMU := $(GOPATH)/bin/promu
-pkgs   = $(shell $(GO) list ./... | grep -v /vendor/)
-
-PREFIX                  ?= $(shell pwd)
-BIN_DIR                 ?= $(shell pwd)
-DOCKER_IMAGE_NAME       ?= event-exporter
-DOCKER_IMAGE_TAG        ?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))
-
-ifdef DEBUG
-	bindata_flags = -debug
-endif
 
+ROOT := github.com/caicloud/event_exporter
+VERSION ?= $(shell git describe --tags --always --dirty)
+COMMIT = $(shell git rev-parse HEAD)
+BRANCH = $(shell git branch | grep \* | cut -d ' ' -f2)
+BUILD_DATE = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 
-all: format build test
+# this is not a public registry; change it to your own
+REGISTRY ?= cargo.dev.caicloud.xyz/release
 
-test:
-	@echo ">> running tests"
-	@$(GO) test -short $(pkgs)
+ARCH ?= amd64
+GO_VERSION = 1.13
 
-style:
-	@echo ">> checking code style"
-	@! gofmt -d $(shell find . -path ./vendor -prune -o -name '*.go' -print) | grep '^'
+CPUS ?= $(shell /bin/bash hack/read_cpus_available.sh)
+GOPATH ?= $(shell go env GOPATH)
+BIN_DIR := $(GOPATH)/bin
+GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
 
-format:
-	@echo ">> formatting code"
-	@$(GO) fmt $(pkgs)
+.PHONY: lint test build build-local container push clean
 
-vet:
-	@echo ">> vetting code"
-	@$(GO) vet $(pkgs)
+build:
+	@docker run --rm -t													\
+	  -v "${PWD}:/go/src/github.com/caicloud/event_exporter"			\
```

Comment:
Format the `\` at the end of lines.

Reply:
they are formatted; you are seeing a difference because here one tab is 8 spaces instead of 4

anyway, I don't think it's much of a big deal. we should move things forward.